.github/workflows/firebase-hosting-merge.yml

```diff
@@ -1,11 +1,8 @@
-# This file was auto-generated by the Firebase CLI
+# This file was initially auto-generated by the Firebase CLI
 # https://github.com/firebase/firebase-tools
 
 name: Deploy to Firebase Hosting on merge
-'on':
-  push:
-    branches:
-      - master
+on: push
```

Comment:
Given that the site only needs the information from the released branches, this can be optimized to avoid re-deploy when PRs merge to master.

Reply:
If there are changes to the site code or the generator, which lives in the main branch, we would want the deployment to pick it up.